composer/tools/composer_db_transfer.md

```diff
@@ -0,0 +1,160 @@
+    newer
+
+1.  The script depends on Python 3.6 (or newer), gcloud and kubectl. Make sure
+    you have all those tools installed.
+
+1.  Make sure that your Composer environment is healthy ("Environment health",
+    "Database health" - see:
+    https://cloud.google.com/composer/docs/monitoring-dashboard). If it is not
+    healthy, first step is to fix the environment.
+
+1.  Make sure that you have write access to the current directory. The script
+    exports KUBECONFIG environment variable to store credentials to the
+    Composer's GKE cluster in current directory (on the machine that executes
+    the script). The purpose is not to interfere with existing GKE-related
+    user's configuration.
+
+1.  Pause the DAGs before exporting the database.
+
+## Limitations
+
+1.  Logs from old DAG executions are not transferred (but history of executions
+    in the database is transferred).
+
+1.  The script does not perform any DAG modifications, does not validate that
+    the DAGs are suitable to run in Airflow 2.0 and does not verify that
+    suitable dependencies are installed in target environment. Airflow
+    [upgrade_check](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html)
+    might be used to troubleshoot DAG compatibility issues.
+
+1.  `dag_code` and `serialized_dag` tables are not synchronized directly.
+    Database entries for imported DAGs are created by Airflow in the new
+    environment.
+
+1.  The script does not prevent DAGs from being scheduled in both old and new
+    environment. If can be prevented e.g. by pausing the DAGs before exporting
+    them and unpausing them in the new environment.
+
+1.  Airflow web server in target environment might temporarily not be available
+    after the database transfer.
+
+1.  The environment might need some time to parse imported DAGs. As a result
+    they might appear in the web UI with some delay.
+
+1.  Import operation removes all records from the target environment. Files
+    existing in environment's bucket are not deleted.
+
+1.  The script does not override gcloud configuration and exports it's own
+    kubeconfig. Temporary files for GKE deployment and kubeconfig created in
+    working directory have unique names.
+
+1.  File with fernet key created by export operation is not deleted by import
+    operation.
+
+## Usage
+
+### Export from an old environment (Airflow 1.10.14/15)
+
+CSV files with exported database are stored as /export/tables/[TABLE NAME].csv
+in the environment's bucket. dags, plugins and data directories are stored in
+/export/dirs in the environment's bucket.
+
+File with a fernet key is going to be created on the machine executing the
+script.
+
+```
+python3 composer_db_transfer.py export \
+    --project [PROJECT NAME] \
+    --environment [ENVIRONMENT NAME] \
+    --location [REGION] \
+    --fernet-key-file [PATH TO FERNET KEY FILE - TO BE CREATED]
```

Comment:
I figured out the answer to the second question! I realize this is the export so it's the old. Might not hurt to update it to say `[EXISTING ENVIRONMENT NAME]`

Reply:
Also, if this fernet-key-file is going to be created, is this just the path on the local machine where we want it to end up?

Reply:
1. They don't need to be in the same project.
2. In both cases (export and import) we operate on existing environments.
3. Fernet key file is created (during export phase) and read from (during import) local machine invoking the script. It is mentioned in "Export from an old environment (Airflow 1.10.14/15)" section.

I updated description of parameters to "SOURCE ENVIRONMENT NAME" / "TARGET ENVIRONMENT NAME". I don't want to use "existing" / "new" wording as it may imply that new environment is created in import phase.